### ISSUE TITLE
[6.2.0] Regression with optional chaining and array (`bar?.baz[0]`)

### ISSUE THREAD

**Issue by regseb** [stryker-mutator/stryker-js#3717]
**Summary**

There is a regression with version 6.2.0. An expression with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and array (`bar?.baz[0]`) do the initial test run fail.

**Files**

- _.stryker.conf.js_

```JavaScript
export default {
    commandRunner: { command: "node test/foo.spec.js" },
};
```

- _package.json_

```JSON
{
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "scripts": {
        "test": "stryker run --fileLogLevel trace"
    },
    "devDependencies": {
        "@stryker-mutator/core": "6.2.0",
        "@stryker-mutator/mocha-runner": "6.2.0"
    }
}
```

- _src/foo.js_

```JavaScript
export default function () {
    const bar = undefined;
    return bar?.baz[0];
};
```

- _test/foo.spec.js_

```JavaScript
import assert from "node:assert";
import foo from "../src/foo.js";

assert.strictEqual(foo(), undefined);
```

**Command**

```shell
npm i
node test/foo.spec.js
npm test
```

**Results**

```
$ npm i

added 268 packages, and audited 269 packages in 4s

65 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
$ node test/foo.spec.js 
$ npm test       

> testcase@1.0.0 test
> stryker run --fileLogLevel trace

16:16:39 (42914) INFO ProjectReader Found 1 of 5 file(s) to be mutated.
16:16:40 (42914) INFO Instrumenter Instrumented 1 source file(s) with 2 mutant(s)
16:16:40 (42914) INFO ConcurrencyTokenProvider Creating 7 test runner process(es).
16:16:40 (42914) INFO DryRunExecutor Starting initial test run (command test runner with "perTest" coverage analysis). This may take a while.
16:16:40 (42914) ERROR DryRunExecutor One or more tests failed in the initial test run:
	All tests
		file:///home/regseb/dev/testcase_ck/.stryker-tmp/sandbox9370320/src/foo.js:70
    return (stryMutAct_9fa48("1") ? bar.baz : (stryCov_9fa48("1"), bar?.baz))[0];
                                                                             ^

TypeError: Cannot read properties of undefined (reading '0')
    at default (file:///home/regseb/dev/testcase_ck/.stryker-tmp/sandbox9370320/src/foo.js:70:78)
    at file:///home/regseb/dev/testcase_ck/.stryker-tmp/sandbox9370320/test/foo.spec.js:5:20
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:528:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)

Node.js v18.8.0

16:16:40 (42914) ERROR Stryker There were failed tests in the initial test run.
```

**Your Environment**

| software         | version(s)
| ---------------- | -------
| node             | v18.8.0
| npm              | 8.18.0
| Operating System | Ubuntu 22.04.1 LTS

**Add stryker.log**

[stryker.log](https://github.com/stryker-mutator/stryker-js/files/9497829/stryker.log)

**Comment by nicojs** [comment]
Thanks for the fast feedback! Regression happened here: #3713. #3718 should fix it.

**Comment by nicojs** [comment]
Fix released in 6.2.2 🚀
